base/strings/util.jl

```diff
@@ -324,8 +324,8 @@ end
 function _split(str::AbstractString, splitter, limit::Integer, keepempty::Bool, strs::Array)
     i = 1 # firstindex(str)
     n = lastindex(str)
-    r = something(findfirst(splitter,str), 0)
-    if r != 0:-1
+    r = findfirst(splitter,str)
+    if !isnothing(r)
```

Comment:
Interesting — it looks like this was a bug previously?  I don't think this changes any behaviors (since the `0` gets cut out by the `while` loop below), but 👍 to fixing this appropriately.

Reply:
It wasn’t a bug in the sense that the code didn’t work. The condition `r != 0:-1` was always `true`, but this did not result in wrong behavior, since the code works correctly when `findfirst` returns numbers (not ranges), and `something(findfirst(splitter,str), 0)` returns `0` when it would have returned `0:-1` in Julia ≤ 0.6.

The code is only faulty when empty ranges compare equal, and only for an empty delimiter (e.g., `split("abc", "")`). In this case, `findfirst` returns empty ranges: `findfirst("", "abc") === 1:0`. These ranges are equal to `0:-1`, leading to wrong results.

Reply:
Ah, I get it.  I was trying to understand if the prior behavior was actually functionally a bug (and thus needed a test) but I see now that it's this change that leads to a bug being exposed — and therefore that's how you found it.